TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc

```diff
@@ -750,6 +755,11 @@ void TrackDetectorAssociator::fillMuon( const edm::Event& iEvent,
    if (! cscSegments.isValid()) 
      throw cms::Exception("FatalError") << "Unable to find CSCSegmentCollection in event!\n";
 
+   edm::Handle<GEMSegmentCollection> gemSegments;
+   iEvent.getByToken(parameters.gemSegmentsToken, gemSegments );
+   edm::Handle<ME0SegmentCollection> me0Segments;
+   iEvent.getByToken(parameters.me0SegmentsToken, me0Segments );
```

Comment:
better call these only if these detectors logic is enabled in the configuration.